composer/distributed/fsdp2.py

```diff
@@ -190,4 +245,14 @@ def prepare_fully_shard(
         None
     """
     modules_to_shard, _ = get_standalone_and_tied_modules(list(model.children()))
+
+    # Build the parameter ID to name mapping (with no duplicates)
+    orig_param_id_to_name = {id(param): name for name, param in model.named_parameters()}
+    for module in modules_to_shard:
```

Comment:
I think you can just make the original call with `recurse=True` and avoid the extra for loop?

Reply:
oh bowen left this comment somewhere else too